src/http/headers.cr

```diff
@@ -275,7 +285,18 @@ struct HTTP::Headers
 
   forward_missing_to @hash
 
+  def serialize(io)
+    @hash.each do |key, values|
+      cast(values).each do |value|
+        io << key.name << ": " << value << "\r\n"
+      end
+    end
+    io << "\r\n"
+    @serialized = true
```

Comment:
Setting the flag should be the first operation in this method to avoid race conditions.

Reply:
You are absolutely right